c7n/offhours.py

```diff
@@ -277,10 +294,41 @@ def get_local_tz(self, parts):
             self.log.warning(
                 "filter:offhours unknown tz %s for %s" % (
                     tz_spec, parts))
-                    
+
             return None
         return tz
-    
+
+
+    def is_custom(self, parts):
+        if len(parts) == 3 and parts.get("on") and parts.get("off"):
+            return True
+        else:
+            return False
+
+    def get_custom_time(self, i, now, parts):
+        if type(self).__name__ in ("InstanceOnHour", "OnHour"):
```

Comment:
Would this be better for this block of code?
Just want to make sure I'm reading the question here correctly.
for ASGs which use Offhours or Onhours the class would still be one of these?
```
if self.__class__.__name__ in ("InstanceOnHour", "OnHour"):
            time = parts.get("on")
elif self.__class__.__name__ in ("InstanceOffHour", "OffHour"):
            time = parts.get("off")
```